src/main/java/org/icgc/argo/program_service/services/ego/EgoService.java

```diff
@@ -277,7 +277,7 @@ public void cleanUpProgram(String programShortName) {
     egoClient.removePolicyByName("PROGRAMDATA-" + programShortName);
   }
 
-  public Boolean joinProgram(@Email String email, String programShortRole, UserRole role) {
+  public Boolean joinProgram(@Email String email, @NonNull String programShortRole, @NonNull UserRole role) {
```

Comment:
Please remove the MailService and JoinProgramInviteRepository dependencies from EgoService. They are no longer used. I cant comment on non-pr changes, which is why im commenting here

Reply:
removed these dependencies in my other pr, hopefully get them all merged today.